### DISCUSSION TITLE
Roles&Permissions on query and mutations

### DISCUSSION THREAD

**Discussion by Levon770** [webonyx/graphql-php#1196] [General]
Can you suggest a good way to handle permissions. Like after custom logic the query returns "permission deny" error, same as 403 error.
Can't find any good idea. Currently the check was done in resolver. But bad thing is that each resolver has to call same function for permission check and return mixed Type. I am looking for idea of middleware, like authantication.

**Comment by le0daniel** [comment]
You could use a Dataloder, so you check for the permission once...
